File: practice/algorithms/implementation/divisible_sum_pairs/DivisibleSumPairs.scala
object DivisibleSumPairs extends App {
  val lines = Source.stdin.getLines().toList
  val divisor = lines(0).split(" ").map(_.toInt).toList(1)
  val array = lines(1).split(" ").map(_.toInt)

  val indexPairs = array.indices.flatMap(i => array.indices.slice(i + 1, array.length).map(j => (i, j)))
  val divisible = indexPairs.count(pair => (array(pair._1) + array(pair._2)) % divisor == 0)
  println(divisible)
}

File: practice/algorithms/implementation/divisible_sum_pairs/DivisibleSumPairs.java
import java.util.*;

public class DivisibleSumPairs {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int n = stdin.nextInt();
        int divisor = stdin.nextInt();
        int[] array = new int[n];
        for(int i = 0; i < n; i++) {
            array[i] = stdin.nextInt();
        }

        int count = 0;
        for(int i = 0; i < n - 1; i++) {
            for(int j = i + 1; j < n; j++) {
                if((array[i] + array[j]) % divisor == 0) {
                    count++;
                }
            }
        }
        System.out.println(count);
        stdin.close();
    }
}